android/app/src/main/kotlin/com/yubico/authenticator/fido/FidoPinStore.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2024 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.authenticator.fido
+
+class FidoPinStore {
+    private var pin: CharArray? = null
+
+    fun hasPin(): Boolean {
+        return pin != null
+    }
+
+    fun getPin(): CharArray {
+        return pin!!
+    }
+
+    fun setPin(newPin: CharArray?) {
+        pin = newPin?.clone()
```

Comment:
Should we zero the old pin here if it is set before re-assigning it?

Reply:
Will fix 👍🏻